spec/requests/slack_post_event_spec.rb

```diff
@@ -0,0 +1,104 @@
+require "rails_helper"
+
+RSpec.describe "Slack post event", type: :request do
+  describe "publishing events to Redis" do
+    it "publishes events even if token is invalid" do
+      expect(REDIS).to receive(:publish).with(
+        :slack_events, { token: "invalid" }.to_json
+      )
+
+      post "/slack/event", params: { slack: { token: "invalid" } }
+    end
+
+    it "sends errors to Rollbar" do
+      error = StandardError.new("oops")
+      allow(REDIS).to receive(:publish).and_raise(error)
+      expect(Rollbar).to receive(:error).with(
+        error, slack_params: { "foo" => "bar" }
+      )
+      post "/slack/event", params: { slack: { foo: "bar" } }
+    end
+  end
+
+  describe "when token is invalid" do
+    it "returns http forbidden" do
+      post "/slack/event", params: { slack: { token: "invalid" } }
+      expect(response).to have_http_status(403)
+    end
+  end
+
+  describe "when token is valid" do
+    let(:token) { "valid" }
+
+    before do
+      allow(ENV).to receive(:[]).with("SLACK_VERIFICATION_TOKEN").and_return(token)
```

Comment:
This makes me a little nervous if only because it will break if I change the code from `[]` to use `fetch`.   Maybe we could pull in https://github.com/ljkbennett/stub_env or maybe we can commit a `.env.test` file that has a fake value defined and reference that?

Reply:
That's a good point. We could add in a `.env.test` with a fake value but I would like to set it explicitly in the tests which gives us the flexibility to stub different values for different specs. I am okay with using in the `stub_env` gem.

What do you think?